scripts/src/main/resources/scripts/functions

```diff
@@ -1824,53 +1830,142 @@ function doGetSoftwareVersion() {
   fi
 }
 
+# $1: software
+function doGetSoftwareEdition() {
+  local software="${1}"
+  local edition
+  echo "${software}" | grep '-' >/dev/null
+  ERG="$?"
+  if [ "${ERG}" = "0" ]
+  then
+    echo "${software}"
+  else
+    edition="$(echo "${software}" | awk '{print toupper($0)}')_EDITION"
+    if [ "$(printf '%s\n' "${!edition}")" == "" ]
+    then
+      echo "${software}"
+    else
+      echo "${!edition}"
+    fi
```

Comment:
I am also confused by this complication. Is there some real rationale behind the `printf` magic and if you append a newline how can it ever be empty?

How about this:
```
function doGetSoftwareEdition() {
  local software="${1}"
  local editionVar
  editionVar="$(echo "${software/-/_}" | awk '{print toupper($0)}')_EDITION"
  local edition="${!editionVar}"
  if [ -z "${edition}" ]
  then
    edition="${software}"
  fi
  echo "${software}"
}
```

Reply:
my code is not tested - just a best guess